### PR TITLE
Add VGGSound classification recipe

### PR DESCRIPTION
**Pull request by xiquan-li**
## What did you change?

Add VGGSound Classification recipe

## Why did you make this change?

Espnet does not support classification on VGGSound 
---

## Is your PR small enough?

---

### CODE REVIEW

**Comment by sw005320** [comment]
Please fix the CI error (e.g., https://github.com/espnet/espnet/actions/runs/27751005994/job/82103382161?pr=6475)

File: egs2/vggsound/cls1/README.md
Comment (sw005320) [inline egs2/vggsound/cls1/README.md:1]: You should merge egs2/vggsound/cls1/RESULTS.md into egs2/vggsound/cls1/README.md

File: egs2/vggsound/cls1/RESULTS.md
Comment (sw005320) [inline egs2/vggsound/cls1/RESULTS.md:1]: Is this performance reasonable?
What is the position of this work compared with the state-of-the-art result (in a similar setup)?

File: egs2/vggsound/cls1/db.sh
Comment (sw005320) [inline egs2/vggsound/cls1/db.sh:1]: what's this?
Many of them are not related to this PR.
We usually use `dp.sh` from the symlink.
Please acheck the other recipe

File: egs2/vggsound/cls1/README.md
Comment (xiquan-li) [reply egs2/vggsound/cls1/README.md:1]: Thanks, merged RESULTS.md into README.md, following egs2/as20k/cls1 and egs2/meld/cls1.

File: egs2/vggsound/cls1/db.sh
Comment (xiquan-li) [reply egs2/vggsound/cls1/db.sh:1]: That was a copied db.sh. Replaced it with a symlink to ../../TEMPLATE/cls1/db.sh, same as as20k and meld, and added VGGSOUND= to egs2/TEMPLATE/asr1/db.sh.

File: egs2/vggsound/cls1/RESULTS.md
Comment (xiquan-li) [reply egs2/vggsound/cls1/RESULTS.md:1]: Test top-1 is 56.07%. 
This is above the original VGGSound CNN baselines (ResNet50 51.0%). 
Stronger audio-only models (CAV-MAE 59.5, MAViL 60.8) and audio-visual SOTA are higher. This recipe is a reproducible BEATs fine-tuning baseline, not a SOTA attempt. valid 77.7% is a random split of train, not an official test set, so it should not be compared with papers. Added this comparison to the README.

**Comment by sw005320** [comment]
@xiquan-li, please also deal with coderabbitai's reviews.
Some of them are reasonable.

File: egs2/vggsound/cls1/conf/beats_cls_vggsound.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# BEATs computes mel features, normalization, and SpecAug internally.
+frontend: none
+input_size: 1
+normalize: none
+
+encoder: beats
+encoder_conf:
+    # Download a BEATs checkpoint and update this path before training.
+    # Example: BEATs_iter3_plus_AS2M.pt from microsoft/unilm BEATs.
+    beats_ckpt_path: PATH/TO/BEATs_iter3_plus_AS2M.pt
```
[the remark this answers is not in the document]
Comment (xiquan-li) [reply egs2/vggsound/cls1/conf/beats_cls_vggsound.yaml:45]: Same as as20k: users download the BEATs ckpt and set the path. No extra download stage.

File: egs2/vggsound/cls1/db.sh [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+SPEECH_PROMPT_v2=
+STOP=
+SWBD=
+FISHER_CALLHOME_SPANISH=
+SWBD_NXT=
+THCHS30=downloads
+TIMIT=
+TSUKUYOMI=downloads
+VOXFORGE=downloads
+VGGSOUND=
```
[the remark this answers is not in the document]
Comment (xiquan-li) [reply egs2/vggsound/cls1/db.sh]: Done: now a symlink, with VGGSOUND= in the template.

File: egs2/vggsound/cls1/local/data_prep_vggsound.py [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+
+    missing = 0
+    written = 0
+    with (
+        open(out_dir / "wav.scp", "w", encoding="utf-8") as wav_f,
+        open(out_dir / "text", "w", encoding="utf-8") as text_f,
+        open(out_dir / "utt2spk", "w", encoding="utf-8") as utt2spk_f,
+    ):
+        for index, item in enumerate(progress(items, f"Writing {split}", len(items))):
+            audio_path = find_audio(root, split, item)
```
[the remark this answers is not in the document]
Comment (xiquan-li) [reply egs2/vggsound/cls1/local/data_prep_vggsound.py]: Fixed: valid clips now look up audio using the source split (train/).

File: egs2/vggsound/cls1/local/data.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+log "$0 $*"
+DATA_PREP_ROOT=${1:-"."}
+shift $(( $# > 0 ? 1 : 0 ))
```
[the remark this answers is not in the document]
Comment (xiquan-li) [reply egs2/vggsound/cls1/local/data.sh]: Fixed: options are no longer parsed as the datadir.

File: egs2/vggsound/cls1/run.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set="train"
+valid_set="valid"
+test_sets="test"
+cls_config=conf/beats_cls_vggsound.yaml
```
[the remark this answers is not in the document]
Comment (xiquan-li) [reply egs2/vggsound/cls1/run.sh:11]: cls1 recipes (as20k, meld) keep the yaml under conf/ directly; this follows that.

**Comment by xiquan-li** [comment]
@sw005320 
Addressed the reasonable CodeRabbit comments (db.sh symlink, data.sh option parsing, valid audio lookup). 
Left the ASR conf/train.yaml layout and automatic BEATs download unchanged, matching as20k/meld.

**Review by sw005320 — APPROVED** [review]
Thanks, @xiquan-li!
